internal/factsengine/adapters/rabbitmq.go

```diff
@@ -52,13 +53,14 @@ func (r *RabbitMQAdapter) Unsubscribe() error {
 	return nil
 }
 
-func (r *RabbitMQAdapter) Listen(agentID string, handle func([]byte) error) error {
+func (r *RabbitMQAdapter) Listen(agentID string, handle func(string, []byte) error) error {
 	return r.consumer.StartConsuming(
 		func(d rabbitmq.Delivery) rabbitmq.Action {
 			// TODO: Handle different kind of errors returning some sort of metadata
 			// so the applied action is potentially changed
-			err := handle(d.Body)
+			err := handle(d.ContentType, d.Body)
```

Comment:
If I understand it correctly, this means the content type `application/cloudevents+json` is expected in the amqp headers, and it will also be part of the body of the message.

An if I get it right it comes handy in order to avoid unmarshalling the body at this point.

It doesn't seem wanda puts that header, at the moment. We need to make the things click at a certain point :smile: